spec/griddler/cloudmailin/adapter_spec.rb

```diff
@@ -1,43 +1,120 @@
 require 'spec_helper'
 
+describe Griddler::Cloudmailin::Adapter do
+  it 'registers itself with griddler' do
+    expect(Griddler.adapter_registry[:cloudmailin]).to be(
+      Griddler::Cloudmailin::Adapter
+    )
+  end
+end
+
 describe Griddler::Cloudmailin::Adapter, '.normalize_params' do
-  it_should_behave_like 'Griddler adapter', :cloudmailin, {
-    envelope: {
-      to: 'Hello World <hi@example.com>',
-      from: 'There <there@example.com>',
-    },
-    plain: 'hi',
-    headers: { Cc: 'emily@example.com' },
-  }
+  context 'without bcc' do
+    it_should_behave_like(
+      'Griddler adapter',
+      :cloudmailin,
+      envelope: {
+        to: 'hi@example.com',
+        from: 'there@example.com'
+      },
+      plain: 'hi',
+      headers: {
+        From: 'there@example.com',
+        To: 'Hello World <hi@example.com>',
+        Cc: 'emily@example.com'
+      }
+    )
+  end
+
+  context 'with bcc' do
+    it_should_behave_like(
+      'Griddler adapter',
+      :cloudmailin,
+      envelope: {
+        to: 'sandra@example.com',
+        from: 'there@example.com'
+      },
+      plain: 'hi',
+      headers: {
+        From: 'there@example.com',
+        To: 'Hello World <hi@example.com>',
+        Cc: 'emily@example.com'
+      }
+    )
+  end
 
   it 'normalizes parameters' do
-    expect(Griddler::Cloudmailin::Adapter.normalize_params(default_params)).to be_normalized_to({
+    expect(normalized).to be_normalized_to(
       to: ['Some Identifier <some-identifier@example.com>'],
       cc: ['emily@example.com'],
       from: 'Joe User <joeuser@example.com>',
       subject: 'Re: [ThisApp] That thing',
       text: /Dear bob/
-    })
+    )
   end
 
-  it 'passes the received array of files' do
-    params = default_params.merge({ attachments: [upload_1, upload_2] })
+  it 'normalizes parameters where the recipient has been BCCed' do
+    expect(normalized(bcc_params)).to be_normalized_to(
+      to: ['Some Identifier <some-identifier@example.com>'],
+      cc: ['emily@example.com'],
+      bcc: ['sandra@example.com'],
+      from: 'Joe User <joeuser@example.com>',
+      subject: 'Re: [ThisApp] That thing',
+      text: /Dear bob/
+    )
+  end
 
-    normalized_params = Griddler::Cloudmailin::Adapter.normalize_params(params)
-    expect(normalized_params[:attachments]).to eq [upload_1, upload_2]
+  it 'normalizes parameters where the recipient has been CCed' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.